### PR TITLE
WT-11233 Ensure all evergreen variants are using the v4 toolchain (#9352) (#9355) (v6.0 backport)

### PR DESCRIPTION
**Pull request by ajmorton**
(cherry picked from commit 3aa2d3f1ccd04e41d1b8d553ed09713a65c61da0) (cherry picked from commit b95a1af76599098cfa9564a12697bf866ec6f04d)

### CODE REVIEW

**Comment by ajmorton** [comment]
There's a few small changes needed to align to the 6.0 branch. I'll call them out in the code diff

File: test/evergreen.yml [hunk 1]
```diff
@@ -141,7 +141,7 @@ functions:
           # Use the Windows powershell script to configure the CMake build.
           # We execute it in a powershell environment as its easier to detect and source the Visual Studio
           # toolchain in a native Windows environment. We can't easily execute the build in a cygwin environment.
-          powershell.exe  -NonInteractive '.\test\evergreen\build_windows.ps1' -configure 1 ${windows_configure_flags|}
+          ${configure_env_vars|} powershell.exe  -NonInteractive '.\test\evergreen\build_windows.ps1' -configure 1 ${windows_configure_flags|}
```
Comment (ajmorton) [inline test/evergreen.yml:144]: `configure_env_vars` and `compile_env_vars` weren't present on 6.0 previously. This backport introduces them as I don't expect to backport the original commit

File: test/evergreen.yml [hunk 2]
```diff
@@ -885,15 +885,16 @@ tasks:
       - func: "compile wiredtiger"
         vars:
           posix_configure_flags: -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/clang.cmake -DENABLE_STRICT=1 -DHAVE_DIAGNOSTIC=0
-      - func: "compile wiredtiger"
```
Comment (ajmorton) [inline test/evergreen.yml:888]: This issue is tracked in WT-11239. Newer branches have dedicated `CLANG_C_VERSION` and `CLANG_CXX_VERSION` flags but they're missing in the `configure wiredtiger` task, effectively not being run.
WT-11239 will fix this but it's uncovered some platform issues with libclang that will also need fixing on 6.0.
The cleanest solution for now is to disable and backport WT-11239

**Review by etienneptl — APPROVED** [review]

File: test/evergreen.yml (hunk 2 above)
Comment (etienneptl) [reply test/evergreen.yml:888]: Sounds good, thank you. Should we prioritise [WT-11239](https://jira.mongodb.org/browse/WT-11239)?

Comment (ajmorton) [reply test/evergreen.yml:888]: I think it's ok right now. I've got the ticket moving in the background and I don't think we're losing much coverage here. We're still testing on both gcc and clang which is the important part. Note as part of WT-11239 I'm also discussing if we should add tests for versions 9 through 15 which we currently don't cover.

**Pull request merged**